src/server/common/model/answer/checkbox/checkbox.js

```diff
@@ -0,0 +1,143 @@
+import Joi from 'joi'
+
+import { NotImplementedError } from '../../../helpers/not-implemented-error.js'
+import { AnswerModel } from '../answer-model.js'
+import { validateAnswerAgainstSchema } from '../validation.js'
+
+/** @import {AnswerViewModelOptions} from '../answer-model.js' */
+
+/**
+ * @param {CheckboxConfig} config
+ * @returns {Joi.Schema}
+ */
+const createCheckboxSchema = (config) => {
+  const optionSchema = Joi.string().valid(...Object.keys(config.options))
+
+  let optionsSchema = Joi.array().items(optionSchema)
+
+  if (config.validation.empty) {
+    const emptyOptionText = config.validation.empty.message
+    optionsSchema = optionsSchema.min(1).messages({
+      'array.min': emptyOptionText
+    })
+  }
+
+  return Joi.object({ [config.payloadKey]: optionsSchema })
+}
+
+/**
+ * @template T
+ * @param {T[] | T} value
+ * @returns T[]
+ */
+const ensureArray = (value) => (Array.isArray(value) ? value : [value])
```

Comment:
could be moved to a helper file?

Reply:
Not sure about that, but we could even allow undefined input and return empty array so we can simplify line 114?

Reply:
@joseluisgraa  - I'd happily do so, if we discover another use case for it?

This is the only type of form input I know of that results in a Hapi body that has either 'string | string[]', which is the use-case we're handling here. 

:thinking:

Reply:
> Not sure about that, but we could even allow undefined input and return empty array so we can simplify line 114?

This is a good suggestion - updating.